src/pages/instances/InstanceSearchFilter.tsx

```diff
@@ -49,86 +42,43 @@ const InstanceSearchFilter: FC<Props> = ({ instances }) => {
       }),
     },
   ];
-  const setSearchParams = useSearchParams()[1];
-  const onSearchDataChange = (searchData: SearchAndFilterChip[]) => {
-    const filteredQueries = searchData
-      .filter((chip) => chip.quoteValue === true || chip.lead === "query")
-      .map((chip) => chip.value.toLowerCase());
-
-    const filteredStatuses = searchData
-      .filter((chip) => chip.lead === "status")
-      .map((chip) => chip.value as LxdInstanceStatus);
-    const filteredTypes = searchData
-      .filter((chip) => chip.lead === "type")
-      .map((chip) => chip.value);
 
-    const profileQueries = searchData
-      .filter((chip) => chip.lead === "profile")
-      .map((chip) => chip.value);
+  const onSearchDataChange = (searchData: SearchAndFilterChip[]) => {
+    const searchTokensByLead = (lead: string) =>
+      searchData.filter((chip) => chip.lead === lead).map((chip) => chip.value);
 
-    const currentWindowSearchParams = new URLSearchParams(
-      window.location.search,
-    );
-    const newSearchParams = new URLSearchParams();
-    filteredQueries.forEach((query) => {
-      newSearchParams.append("q", query);
+    const newParams = new URLSearchParams();
+    searchTokensByLead("quote").forEach((query) => {
+      newParams.append("q", query);
```

Comment:
You renamed the lead to `quote`, but kept the value of the `forEach` argument as `query`. This is quite confusing. I advise to keep a consistent naming.

Reply:
`searchTokensByLead` gives the tokens for a given lead. In this exampl ethe lead is quote, so it returns all requies. In the forEach we iterate over all queries and each item is a query.

Reply:
The lead is quote because you renamed it to quote. It used to be named query, and in that case the name was consistent. Now it is not.

Reply:
Ah, right. I bamboozled the two strings. Will fix it shortly.

Reply:
It's really a nitpick because the lead is invisible for this group of filters, but it left my brain confused for some seconds when reading the code because I was expecting a consistency between the lead name and the argument name of the `forEach` function.

Reply:
Yes and there was a bug, when you refreshed, the queries would get lost, they need to be without lead and have the `quoteValue:true` parameter instead.